src/components/big-interactive-pages/editor.tsx

```diff
@@ -62,89 +64,123 @@ const foldAllTemplateLiterals = () => {
 	);
 };
 
-let lastSavePromise = Promise.resolve();
-let saveQueueSize = 0;
-export const saveGame = debounce(
-	800,
-	(persistenceState: Signal<PersistenceState>, code: string) => {
-		const doSave = async () => {
-			const attemptSaveGame = async () => {
-				try {
-					const game =
-						persistenceState.value.kind === "PERSISTED" &&
-						persistenceState.value.game !== "LOADING"
-							? persistenceState.value.game
-							: null;
-					const res = await fetch("/api/games/save", {
-						method: "POST",
-						headers: { "Content-Type": "application/json" },
-						body: JSON.stringify({
-							code,
-							gameId: game?.id,
-							tutorialName: game?.tutorialName,
-							tutorialIndex: game?.tutorialIndex,
-						}),
-					});
-					if (!res.ok)
-						throw new Error(
-							`Error saving game: ${await res.text()}`
-						);
-					return true;
-				} catch (error) {
-					console.error(error);
-
-					persistenceState.value = {
-						...persistenceState.value,
-						cloudSaveState: "ERROR",
-					} as any;
-					return false;
-				}
-			};
+let lastSavePromise = Promise.resolve()
+let saveQueueSize = 0
 
-			while (!(await attemptSaveGame())) {
-				await new Promise((resolve) => setTimeout(resolve, 2000)); // retry saving the game every 2 seconds
-			}
+export const saveGame = debounce(800, async (persistenceState: Signal<PersistenceState>, code: string, sessionId: string) => {
```

Comment:
Understandable, it was because the branch I was very old and I had to merge but it overwrite other people changes so had to manually re check and add stuff that was from main branch , u can check the last 2 commit before the branch merge to see what's my changes